rfcs/0013-synthea-future-claims-and-automation.md

```diff
@@ -0,0 +1,139 @@
+
+Motivation:
+As previous synthetic data releases were separated by 3 months, and only 20,000 synthetic beneficiaries released, data became quickly outdated, and the amount of data being released doesn't justify the usecase of synthetic data. Additionally, there is inefficiency involved with producing and releasing synthetic data when done without automation. Manual process are prone to human error, and can cause incidents, such as accidental overlap of beneficiary IDs between different synthetic data batches. There are more than enough reasons to look into streamlining and scaling up synthetic data production to maximize the benefit to users and partners. This involves automated recurring Synthea data generation, automated recurring loading, large Synthea data generation, and potential required changes to the BFD server and pipeline application.
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+Automated Recurring Generation: 
+To increase the benefit and quality of synthetic data to the user, data that is current, or for future use, needs to be generated on a more regular basis. The proposed plan is weekly and quarterly. 
+
+  - Weekly - Synthetic data will be generated with a set batch size using the master branch of the Synthea codebase in a hosted cloud instance, along with the Synthea end-state properties file from the most recent generated batch. The end-state properties file is critical for making sure the next batch of Synthea data will not overlap with released Synthea data.
+
+  - Quarterly - The main difference in the process of generating Synthea data each quarter will be using the most recent version of the master branch of the Synthea codebase. This should introduce contrast in beneficiary and claim data properites between quarterly and weekly generated data, as Mitre regularly makes changes to the output of Synthea data generated over time. 
+
+  - Future Claim Data - For both quarterly and weekly, a batch of similar size for claim data with a future date will also be generated in a similar fashion. 
+
+  - Special data parameters of interest: One parameter, which has not changed in previous releases of Synthea is Part D Event ID (PDE ID). Despite monotonically increasing other parameters i.e. bene ID, claim group ID, the value of PDE ID will not need to change on a regular basis, unless there is a specific customer use case presented
+
+Automated Recurring Loading: 
+  - Once the synthetic data is generated, within the same hosted cloud instance, the latest version of the BFD application from github, and dockerized database, will have been installed prior, and will be used to run the BFD pipeline integration test (IT), to load the new batch of synthetic data, ensuring that the newly generated data can be inserted properly.
+
+  - Additionally, the RIF files from the previous weekly/quarterly batch of Synthea data will be pulled from AWS S3, and loaded into the database with the new batch of synthea data.  
```

Comment:
I don't quite get this part. Wouldnt the weekly/quarterly batches have been loaded already, if this automated process is put in place? Or is this just filling a test DB with that data?

Reply:
This is replicating what we have been doing manually locally before deploying data to TEST, PROD SBX, or PROD. If we can't load generated data locally, or cause collisions, we dont deploy it. So if we are to automate this testing step, a provisioned EC2 instance we use to generate synthea data and load into a dockerized BFD DB is synonymous to a local environment where we'd do the same manually.